### PR TITLE
Display Default Values in UI

### PR DESCRIPTION
**Pull request by dsbegnoche**
Display Default values in UI on New Items.  

Also, fix for empty Radio/Dropdown numbers via schemas.service.ts and content-field.component.html

### CODE REVIEW

File: src/Squidex/app/shared/services/schemas.service.ts [hunk 1]
```diff
@@ -753,272 +802,272 @@ export class SchemasService {
...
-                .do(() => {
-                    this.analytics.trackEvent('Schema', 'FieldDeleted', appName);
-                })
-                .pretifyError('Failed to delete field. Please reload.');
+            .do(() => {
+                this.analytics.trackEvent('Schema', 'FieldDeleted', appName);
+            })
+            .pretifyError('Failed to delete field. Please reload.');
     }
 }
```
Comment (dsbegnoche) [inline src/Squidex/app/shared/services/schemas.service.ts]: Damn. Formatting issue for some reason. Will commit revert of that :(

File: src/Squidex/app/shared/services/schemas.service.ts [hunk 2]
```diff
@@ -333,6 +337,8 @@ export abstract class FieldPropertiesDto {
     public abstract formatValue(value: any): string;
 
     public abstract createValidators(isOptional: boolean): ValidatorFn[];
+
+    public abstract getDefaultValue(): any;
```
Comment (SebastianStehle) [inline src/Squidex/app/shared/services/schemas.service.ts]: A 5 minute request: Can you make this method not abstract, then we could get rid of a lot of dummy implementations.

File: src/Squidex/app/shared/services/schemas.service.ts [hunk 3]
```diff
@@ -473,6 +491,17 @@ export class DateTimeFieldPropertiesDto extends FieldPropertiesDto {
 
         return validators;
     }
+
+    public getDefaultValue(): any {
+        if (this.calculatedDefaultValue != null) {
+            let now = new Date();
+            if (this.calculatedDefaultValue === 'Today') {
+                now = new Date(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate());
```
Comment (SebastianStehle) [inline src/Squidex/app/shared/services/schemas.service.ts]: The implementation is different from the server implementation where we would only return dates for Today (no Time). Would be good to create some tests for it (and use DateTime class).

File: src/Squidex/app/shared/services/schemas.service.ts (hunk 2 above)
Comment (dsbegnoche) [reply src/Squidex/app/shared/services/schemas.service.ts]: Good point. Fixed.

**Comment by SebastianStehle** [comment]
@dsbegnoche Have you seen my other comment with DateTime Field?

**Comment by dsbegnoche** [comment]
Just now. Forgot about the DateTime class, so looking into that.

**Comment by SebastianStehle** [comment]
I improved the tests, also in the API code (C#). Basically we made the same mistake:

I think this is a bad test, because it is hard to say, what you actually expect:

      expect(dto.getDefaultValue()).toEqual(now.toUTCStringFormat('ddd, DD MMM YYYY hh:mm:ss Z'));

I made the same mistake in the C# code:

      Assert.Equal(Now.ToString().Substring(10), DefaultValueFactory.CreateDefaultValue(field, Now));

Actually I meant SubString(0, 10) or "yyyy-MM-dd". I made it more explicit.

**Comment by dsbegnoche** [comment]
Great point. Thank you for the improvements

**Comment by SebastianStehle** [comment]
Thanks for the PR :)

**Pull request merged**